x/pagination.go

```diff
@@ -0,0 +1,45 @@
+package x
+
+import (
+	"net/http"
+	"strconv"
+)
+
+const paginationMaxItems = 500
+
+// ParsePagination parses limit and page from *http.Request with given limits and defaults.
+func ParsePagination(r *http.Request) (page, limit int) {
+	if offsetParam := r.URL.Query().Get("page"); offsetParam == "" {
```

Comment:
Can be refactored into something like

```suggestion
	const maxIntVal = int(^uint(0) >> 1)
```
```suggestion
	offsetParam := paginationParam(r.URL.Query().Get("page"), 0, maxIntVal)
	limitParam  := paginationParam(r.URL.Query().Get("limit"), paginationMaxItems, paginationMaxItems)
```

```suggestion
func parseParam(rawParamValue string, defaultValue int, maxValue int) (paramVal int) {
	paramVal := defaultValue
	if rawParamValue == "" {
		return
	} 

	if paramVal64, err := strconv.ParseInt(rawParamValue, 10, 64); err != nil {
		paramVal = defaultValue
	} else {
		paramVal = int(paramVal64)
	}
	
	if paramVal > maxValue {
		paramVal = maxValue
	}

	if paramVal < 0 {
		paramVal = defaultValue
	}
	return
}
```

Would be even better if the default for the "limit" is smaller than the maxValue for the limit as having smaller limits gives users the ability to bump up(to the maximum) when needed. Users would likely never set the limits lower than the default in practice.

Reply:
All your remarks should have been addressed now :)

Reply:
Yes, thanks. There's still scope for exacting a function to avoid code duplication.